src/Components/Tasklist.jsx

```diff
@@ -2,11 +2,19 @@ import React from "react";
 import TaskItem from "./TaskItem";
 
 const TaskList = ({ isCompleted, tasks }) => {
-  const pending = tasks.map((task) => {
-    return !task.isCompleted && <TaskItem value={task} />;
+  const pending = tasks.map((task, index) => {
+    return (
+      !task.isCompleted && (
+        <TaskItem key={index} taskIndex={index} value={task} />
+      )
+    );
   });
-  const completed = tasks.map((task) => {
-    return task.isCompleted && <TaskItem value={task} />;
+  const completed = tasks.map((task, index) => {
+    return (
+      task.isCompleted && (
+        <TaskItem key={index} taskIndex={index} value={task} />
+      )
+    );
   });
   return isCompleted ? <div>{completed}</div> : <div>{pending}</div>;
```

Comment:
Why there is a need to create two different variables and loop tasks twice when either Completed or Pending will be passed as Prop. My suggestion is you can replace the whole tasklist logic with a tasks loop as
```js
return (
  tasks.map((task, index) => isCompleted === task.isCompleted && (
      <TaskItem key={index} taskIndex={index} value={task} />
    )
  );
)
```